src/api/lib/tasks/dev/requests.rake

```diff
@@ -199,5 +199,333 @@ namespace :dev do
+            proj_repo.path_elements.find_or_initialize_by(link: target_repository)
+
+            if repository['arch'].is_a?(Array)
+              repository['arch'].each do |architecture|
+                proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: architecture))
+              end
+            else
+              proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: repository['arch']))
+            end
+            proj_repo.save!
+            project.store(comment: "Added #{proj_repo.name} repository")
+          end
+
+          next
+        elsif repository['path']
+          project = Project.find_or_create_by(name: repository['path']['project'])
+          proj_repo = project.repositories.find_or_initialize_by(name: repository['path']['repository'])
+        else
+          next
+        end
+        target_repo = repository['name']
+        target_repository = Repository.find_by_project_and_name(target_project.name, target_repo)
+        target_repository = create(:repository, project: target_project, name: target_repo) if target_repository.nil?
+        proj_repo.path_elements.find_or_initialize_by(link: target_repository)
+
+        if repository['arch'].is_a?(Array)
+          repository['arch'].each do |architecture|
+            proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: architecture))
+          end
+        else
+          proj_repo.repository_architectures.find_or_initialize_by(architecture: Architecture.find_by(name: repository['arch']))
+        end
+        proj_repo.save!
+        project.store(comment: "Added #{proj_repo.name} repository")
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    # rubocop:disable Metrics/PerceivedComplexity
+    def create_users_and_groups_from_meta(request_data:, users:)
+      print_message('creating users and groups from _meta ....')
+      # create users
+      if request_data['person'].present?
+        request_data['person'].each do |person|
+          find_user(person['userid'], users)
+        end
+      end
+
+      return if request_data['group'].nil?
+
+      if request_data['group'].is_a?(Array)
+        request_data['group'].each do |group|
+          next if Group.find_by(title: group['groupid'])
+
+          create(:group, title: group['groupid'])
+        end
+      else
+        group = Group.find_by(title: request_data['group']['groupid'])
+
+        create(:group, title: request_data['group']['groupid']) if group.nil?
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
+
+    def find_user(login, users = [])
+      return if login.nil?
```

Comment:
I don't think this is possible...

Reply:
This condition is to prevent executing this method because in the creation of the review, I'm using this method and a review can be by_user, by_project, by_package etc. That's why I have this check in case by_user is nil

Reply:
To me it sounds like you should handle this there and not here...